src/ts/Signin/SigninForm.spec.tsx

```diff
@@ -0,0 +1,111 @@
+import 'mocha';
+import * as React from 'react';
+import * as chai from 'chai';
+import { shallow } from 'enzyme';
+import * as sinon from 'sinon';
+require('sinon-stub-promise')(sinon);
+import * as sinonChai from 'sinon-chai';
+chai.use(sinonChai);
+const expect = chai.expect;
+import { formsyContext } from '../helpers/test-theme';
+import { browserHistory } from 'react-router';
+import * as api from './api';
+import SigninForm from './SigninForm';
+
+const user = {
+  name: 'Viking Garage',
+  email: 'viking.garage.app@gmail.com',
+  password1: 'secret',
+  password2: 'secret',
+  birthday: new Date(),
+  agree: true,
+}
+
+const response = {
+  user,
+  token: 'uuid-token',
+}
+
+describe('Signin: <SigninForm />', () => {
+  let wrapper;
+  let instance;
+  let push;
+  beforeEach(() => {
+    wrapper = shallow(<SigninForm />, formsyContext());
+    instance = wrapper.instance();    
+    push = sinon.stub(browserHistory, 'push', () => {});
+  });
+
+  afterEach(() => {
+    push.restore();
+  });
+
+  it('check for inner components', () => {
+    expect(wrapper.find('FormsyText')).to.have.length(4);
+    expect(wrapper.find('[name="name"]')).to.have.length(1);
+    expect(wrapper.find('[name="email"]')).to.have.length(1);
+    expect(wrapper.find('[name="password1"]')).to.have.length(1);
+    expect(wrapper.find('[name="password2"]')).to.have.length(1);
+
+    expect(wrapper.find('FormsyDate')).to.have.length(1);
+    expect(wrapper.find('[name="birthday"]')).to.have.length(1);
+
+    expect(wrapper.find('FormsyCheckbox')).to.have.length(1);
+    expect(wrapper.find('[name="agree"]')).to.have.length(1);
+   
+    expect(wrapper.find('button.submit')).to.have.length(1);
+  });
+
+  it('check if submit action gets user info properly', () => {
+    const signin = sinon.stub(api, 'signin', (params) => {
+      expect(params).to.be.deep.equal(user)
+      return Promise.resolve('ok');
+    });
+    instance['submit'](user);
+    signin.restore();
```

Comment:
We are using `sinon-stub-promise` to have synchronous way of stubbing out promises, here are the details https://github.com/substantial/sinon-stub-promise